app/scripts/models/reliers/oauth.js

```diff
@@ -170,13 +170,22 @@ define(function (require, exports, module) {
       }
     },
 
-    _setupOAuthRPInfo () {
-      var clientId = this.get('clientId');
+    _setupOAuthRPInfo (verificationFlow) {
+      const clientId = this.get('clientId');
+      // get the app provided redirect uri
+      const queryRedirectUri = this.get('redirectUri');
 
       return this._oAuthClient.getClientInfo(clientId)
         .then((serviceInfo) => {
-          var result = Transform.transformUsingSchema(
+          const result = Transform.transformUsingSchema(
             serviceInfo, CLIENT_INFO_SCHEMA, OAuthErrors);
+
+          // verification flow doesn't have a redirect uri, so there is nothing to validate
```

Comment:
@rfk thoughts on this plot twist?

Reply:
This wasn't clear from the discussion in #5827 but IIUC, clients are not *required* to provide the redirect_uri in the query parameters, but *if* they provide it then we should validate it and error out on mismatch.  So the logic would be something like:

```
if (queryRedirectUri && queryRedirectUri !== result.redirectUri) {
  throw OAuthErrors.toError('INCORRECT_REDIRECT');
}
```

Which should work OK with the verification flow where it's not provided?

Or, am I misunderstanding something that means clients *must* provide the redirect_uri in the query string for other reasons?